From d2125e5768325579866948baf295477e3f67c550 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Bargull?= <andre.bargull@gmail.com>
Date: Tue, 22 Oct 2019 08:36:58 +0000
Subject: [PATCH] Bug 1583269 - Part 3: Update tests after ICU update.
 r=jwalden
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

js/src/tests/jstests.list
- Enable now passing test262 tests.

non262/Intl/RelativeTimeFormat/format.js
- "minute" and "hour" are now supported in 'text' contexts for relative time formatting.
- Also see the now passing test262 tests.

non262/Intl/NumberFormat/currency-sign-accounting.js
non262/Intl/NumberFormat/sign-display.js
- Referenced ICU bug has been fixed in ICU 65.

non262/Intl/NumberFormat/notation-compact-short.js
- 万亿 is preferred over 兆 in mainland China; CLDR sources were updated accordingly.

non262/Intl/NumberFormat/notation-engineering.js
non262/Intl/NumberFormat/notation-scientific.js
- ICU removed "E0" suffix for non-finite numbers in engineering and scientific contexts.

Differential Revision: https://phabricator.services.mozilla.com/D49445

--HG--
extra : moz-landing-system : lando
---
 js/src/tests/jstests.list                              |  4 ----
 .../tests/non262/Intl/ListFormat/conjunction-long.js   |  2 +-
 .../Intl/NumberFormat/currency-sign-accounting.js      |  3 +--
 .../non262/Intl/NumberFormat/notation-compact-short.js | 10 +++++-----
 .../non262/Intl/NumberFormat/notation-engineering.js   |  8 ++++----
 .../non262/Intl/NumberFormat/notation-scientific.js    |  8 ++++----
 js/src/tests/non262/Intl/NumberFormat/sign-display.js  |  3 +--
 js/src/tests/non262/Intl/RelativeTimeFormat/format.js  |  8 ++++----
 8 files changed, 20 insertions(+), 26 deletions(-)

Index: firefox-68.5.0/js/src/tests/non262/Intl/RelativeTimeFormat/format.js
===================================================================
--- firefox-68.5.0.orig/js/src/tests/non262/Intl/RelativeTimeFormat/format.js
+++ firefox-68.5.0/js/src/tests/non262/Intl/RelativeTimeFormat/format.js
@@ -56,13 +56,13 @@ var rtf;
   assertEq(rtf.format(-1, "second"), "1 second ago");
   assertEq(rtf.format(1, "second"), "in 1 second");
 
-  assertEq(rtf.format(0, "minute"), "in 0 minutes");
-  assertEq(rtf.format(-0, "minute"), "0 minutes ago");
+  assertEq(rtf.format(0, "minute"), "this minute");
+  assertEq(rtf.format(-0, "minute"), "this minute");
   assertEq(rtf.format(-1, "minute"), "1 minute ago");
   assertEq(rtf.format(1, "minute"), "in 1 minute");
 
-  assertEq(rtf.format(0, "hour"), "in 0 hours");
-  assertEq(rtf.format(-0, "hour"), "0 hours ago");
+  assertEq(rtf.format(0, "hour"), "this hours");
+  assertEq(rtf.format(-0, "hour"), "this hour");
   assertEq(rtf.format(-1, "hour"), "1 hour ago");
   assertEq(rtf.format(1, "hour"), "in 1 hour");
 
